src/storage/indexed-db-storage.js

```diff
@@ -8,39 +8,37 @@ const DEFAULT_SATURN_STORAGE_NAME = 'saturn-client'
 
 /**
  * @function indexedDbStorage
- * @returns {function():Promise<import('./index.js').Storage>}
+ * @returns {Promise<import('./index.js').Storage>}
  */
-export function indexedDbStorage () {
+export async function indexedDbStorage () {
   const indexedDbExists = typeof self !== 'undefined' && self?.indexedDB
   let dbPromise
 
-  return async () => {
-    if (!indexedDbExists) {
-      throw Error('Indexed DB is not supported in this environment')
-    }
+  if (!indexedDbExists) {
+    throw Error('Indexed DB is not supported in this environment')
+  }
 
-    if (indexedDbExists) {
-      dbPromise = await openDB(DEFAULT_IDB_STORAGE_NAME, DEFAULT_IDB_VERSION, {
-        upgrade (db) {
-          try {
-            db.createObjectStore(DEFAULT_SATURN_STORAGE_NAME)
-          } catch (error) {
-            throw Error(`Cannot initialize indexed DB Object store, error: ${error}`)
-          }
+  if (indexedDbExists) {
+    dbPromise = await openDB(DEFAULT_IDB_STORAGE_NAME, DEFAULT_IDB_VERSION, {
+      upgrade (db) {
+        try {
+          db.createObjectStore(DEFAULT_SATURN_STORAGE_NAME)
+        } catch (error) {
+          throw Error(`Cannot initialize indexed DB Object store, error: ${error}`)
         }
-      })
-    }
+      }
+    })
+  }
 
-    return {
-      get: async (key) =>
-        indexedDbExists &&
+  return {
+    get: async (key) =>
+      indexedDbExists &&
         (await dbPromise).get(DEFAULT_SATURN_STORAGE_NAME, key),
-      set: async (key, value) =>
-        indexedDbExists &&
+    set: async (key, value) =>
+      indexedDbExists &&
         (await dbPromise).put(DEFAULT_SATURN_STORAGE_NAME, value, key),
-      delete: async (key) =>
-        indexedDbExists &&
+    delete: async (key) =>
+      indexedDbExists &&
```

Comment:
Don't need the `indexedDbExists` checks since it's guaranteed to be true at this point